src/Polly.Core/Simmy/MonkeyStrategyOptions.TResult.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace Polly.Simmy;
+
+#pragma warning disable CS8618 // Required members are not initialized in constructor since this is a DTO, default value is null
+
+/// <summary>
+/// The options associated with the <see cref="MonkeyStrategyOptions"/>.
+/// </summary>
+/// <typeparam name="TResult">The type of result the retry strategy handles.</typeparam>
+public abstract class MonkeyStrategyOptions<TResult> : ResilienceStrategyOptions
+{
+    /// <summary>
+    /// Gets or sets the injection rate for a given execution, which the value should be between [0, 1].
+    /// </summary>
+    /// <remarks>
+    /// Defaults to <see langword="null"/>. Either <see cref="InjectionRateGenerator"/> or this property is required.
+    /// When this property is <see langword="null"/> the <see cref="InjectionRateGenerator"/> is used.
+    /// </remarks>
+    [Range(MonkeyStrategyConstants.MinInjectionThreshold, MonkeyStrategyConstants.MaxInjectionThreshold)]
+    public double? InjectionRate { get; set; }
+
+    /// <summary>
+    /// Gets or sets the injection rate generator for a given execution, which the value should be between [0, 1].
+    /// </summary>
+    /// <remarks>
+    /// Defaults to <see langword="null"/>. Either <see cref="InjectionRate"/> or this property is required.
+    /// When this property is <see langword="null"/> the <see cref="InjectionRate"/> is used.
+    /// </remarks>
+    public Func<ResilienceContext, ValueTask<double>> InjectionRateGenerator { get; set; }
+
+    /// <summary>
+    /// Gets or sets the enable generator that indicates whether or not the chaos strategy is enabled for a given execution.
+    /// </summary>
+    /// <remarks>
+    /// Defaults to <see langword="null"/>. Either <see cref="Enabled"/> or this property is required.
+    /// When this property is <see langword="null"/> the <see cref="Enabled"/> is used.
+    /// </remarks>
+    public Func<ResilienceContext, ValueTask<bool>> EnabledGenerator { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value that indicates whether or not the chaos strategy is enabled for a given execution.
+    /// </summary>
+    /// <remarks>
+    /// Defaults to <see langword="null"/>. Either <see cref="EnabledGenerator"/> or this property is required.
+    /// When this property is <see langword="null"/> the <see cref="EnabledGenerator"/> is used.
+    /// </remarks>
+    public bool? Enabled { get; set; }
+
+    /// <summary>
+    /// Gets or sets the Randomizer generator instance that is used to evaluate the injection rate.
+    /// </summary>
+    /// <remarks>
+    /// The default randomizer is thread safe and returns values between 0.0 and 1.0.
+    /// </remarks>
+    [Required]
+    public Func<double> Randomizer { get; set; } = RandomUtil.Instance.NextDouble;
```

Comment:
Randomizer is exposed on `CompositeStrategyBuilderBase.Randomizer`. Having it here feels redundant.

Reply:
Good point, I didn't realize it, however, it might not be intuitive in the sense that the consumers would need to recall to pass it when instantiating the `CompositeStrategyBuilder`, the advantage of having it in the `MonkeyStrategyOptions` is that it is more visible I'd say.

```c#
  var builder = new CompositeStrategyBuilder
  {
      Randomizer = () => 0.5
  };
  builder.AddBehavior(new BehaviorStrategyOptions
  {
      Enabled = true,
      InjectionRate = 1,
      BehaviorAction = (_) => default
  });
```

vs

```c#
  new CompositeStrategyBuilder().AddBehavior(new BehaviorStrategyOptions
  {
      Enabled = true,
      InjectionRate = 1,
      BehaviorAction = (_) => default,
      Randomizer = () => 0.5
  });
```
Thoughts?

Reply:
True, can you make it optional thought?

 If not specified, it can be be injected fun the builder?

Alternatively, I can also make it exposed on retry options and drop it from the builder.

What would be your preference/suggestion?